bot/skills/covid_mode.py

```diff
@@ -300,21 +309,28 @@ def random_cough(bot: Bot, queue: JobQueue):
         # todo: move "_get_username" to commons
         full_name = _get_username(user)
 
-        if 'infected_since' in user:
+        lethaled = _db.is_lethaled(user['_id'])
+
+        if 'infected_since' in user and 'cured_since' not in user and lethaled is None:
             chance = RANDOM_COUGH_INFECTED_CHANCE
-            days_count = (datetime.now() - user['infected_since']).days
-            if _rng <= LETHALITY_RATE * (days_count ** days_count):
+            delta_seconds = (datetime.now() - user['infected_since']).total_seconds()
+            delta_days_float = delta_seconds / (60 * 60 * 24)
+
+            if _rng <= RANDOM_CURE_RATE ** (2 - delta_days_float):
+                chance = .0
+                _db.cure(user['_id'])
+                message += f"{full_name} излечился от коронавируса\n"
+
+            if _rng <= LETHALITY_RATE * (delta_days_float ** delta_days_float):
                 chance = .0
                 try:
-                    lethaled = _db.is_lethaled(user['_id'])
-                    if lethaled is None:
-                        _db.add_lethality(user['_id'], datetime.now())
-                        bot.restrict_chat_member(get_group_chat_id(), user['_id'],
-                                                 can_add_web_page_previews=False,
-                                                 can_send_media_messages=False,
-                                                 can_send_other_messages=False,
-                                                 can_send_messages=False)
-                        message += f"{full_name} умер от коронавируса, F\n"
+                    _db.add_lethality(user['_id'], datetime.now())
+                    bot.restrict_chat_member(get_group_chat_id(), user['_id'],
```

Comment:
⚠️ this code will lead to failure on python-telegram-bot version 12.4.2 and above. Use `skills.mute.mute_user_for_time` or declaration like: 
```
update.message.reply_text(f"Таймаут для {user.full_name} на {mute_duration}")
        mute_perm = ChatPermissions(
            can_add_web_page_previews=False,
            can_send_media_messages=False,
            can_send_other_messages=False,
            can_send_messages=False
        )
        context.bot.restrict_chat_member(update.effective_chat.id, user.id, mute_perm, until)
```